src/backend/api/meals.js

```diff
@@ -3,12 +3,68 @@ const router = express.Router();
 const knex = require('../database');
 
 router.get('/', async (req, res) => {
+  let meals = knex.select('meal.*').from('meal');
+
+  if ('maxPrice' in req.query) {
+    meals = meals.whereBetween('price', [0, req.query.maxPrice]);
+  }
+
+  if ('availableReservations' in req.query) {
+    meals = meals
+      .join('reservation', 'reservation.meal_id', '=', 'meal.id')
+      .groupBy('meal.id');
+    if (req.query.availableReservations === 'true') {
+      meals = meals.having(
+        'meal.max_reservations',
+        '>',
+        knex.raw('SUM(reservation.number_of_guests)')
+      );
+    } else {
+      meals = meals.having(
+        'meal.max_reservations',
+        '=',
+        knex.raw('SUM(reservation.number_of_guests)')
+      );
+    }
+  }
+
+  if ('title' in req.query) {
+    meals = meals.whereRaw('LOWER(title) LIKE ?', [`%${req.query.title}%`]);
+  }
+
+  if ('dateAfter' in req.query) {
+    meals = meals.whereRaw('`when` > ?', [req.query.dateAfter]);
```

Comment:
Looks like this works but here's a simpler way to do it:

```suggestion
    meals = meals.where('when', '>', req.query.dateAfter);
```

Same applies for the `dateBefore` parameter.